src/core/constraint.jl

```diff
@@ -37,11 +66,61 @@ end
 
 ""
 function constraint_active_branch(pm::_PMs.AbstractPowerModel,  i::Int, nw_1::Int, nw_2::Int)
-    if haskey(_PMs.ref(pm, nw_1, :branch_damaged), i)
+    if haskey(_PMs.ref(pm, nw_1, :damaged_branch), i)
         z_branch_1 = _PMs.var(pm, nw_1, :z_branch, i)
         z_branch_2 = _PMs.var(pm, nw_2, :z_branch, i)
 
         JuMP.@constraint(pm.model, z_branch_2 >= z_branch_1)
     end
 end
 
+"Load delivered at each node must be greater than or equal the previous time period"
+function constraint_increasing_load(pm::_PMs.AbstractPowerModel,  i::Int, nw_1::Int, nw_2::Int)
+    if haskey(_PMs.ref(pm, nw_1, :damaged_branch), i)
+        z_demand_1 = _PMs.var(pm, nw_1, :z_demand, i)
+        z_demand_2 = _PMs.var(pm, nw_2, :z_demand, i)
+
+        JuMP.@constraint(pm.model, z_demand_2 >= z_demand_1)
+    end
+end
+
+
+"on/off constraint for generators connected to damaged buses"
+function constraint_gen_bus_connection(pm::_PMs.AbstractPowerModel, n::Int, c::Int, gen_id::Int, bus_id::Int, pmin, pmax, qmin, qmax)
+    pg = _PMs.var(pm, n, c, :pg, gen_id)
+    qg = _PMs.var(pm, n, c, :qg, gen_id)
+    z_bus = _PMs.var(pm, n, :z_bus, bus_id)
+
+    JuMP.@constraint(pm.model, pg <= pmax*z_bus)
+    JuMP.@constraint(pm.model, pg >= pmin*z_bus)
+    JuMP.@constraint(pm.model, qg <= qmax*z_bus)
+    JuMP.@constraint(pm.model, qg >= qmin*z_bus)
+end
+
+
+"on/off constraint for generators"
+function constraint_gen_bus_connection(pm::_PMs.AbstractDCPModel, n::Int, c::Int, gen_id::Int, bus_id::Int, pmin, pmax, qmin, qmax)
+    pg = _PMs.var(pm, n, c, :pg, gen_id)
+    z_bus = _PMs.var(pm, n, :z_bus, bus_id)
+
+    JuMP.@constraint(pm.model, pg <= pmax*z_bus)
+    JuMP.@constraint(pm.model, pg >= pmin*z_bus)
+end
+
+function constraint_voltage_magnitude_on_off(pm::_PMs.AbstractPowerModel, n::Int, c::Int, i::Int, vmin, vmax)
+    vm = _PMs.var(pm, n, c, :vm, i)
+    z_bus = _PMs.var(pm, n, :z_bus, i)
+
+    JuMP.@constraint(pm.model, vm <= vmax*z_bus)
+    JuMP.@constraint(pm.model, vm >= vmin*z_bus)
+end
+
+function constraint_voltage_magnitude_sqr_on_off(pm::_PMs.AbstractPowerModel, n::Int, c::Int, i::Int, vmin, vmax)
```

Comment:
The type signature is not correct here.  This implementation only applies to `W` models.